### PR TITLE
Fix issue when filtering nested array

### PR DESCRIPTION
**Pull request by tiopramayudi**
We found bugs where the jsonpath is not working as expectedly when dealing with nested array. Suppose you have
```
{
    "inputs": [
        {
            "variable": [
                {
                    "name": "A",
                    "value": 200
                },
                {
                    "name": "B",
                    "value": 250
                }
            ]
        },
        {
            "variable": [
                {
                    "name": "C",
                    "value": 300
                },
                {
                    "name": "D",
                    "value": 350
                }
            ]
        }
    ]
}
```
This path `$.inputs[*].variable[?(@.value > 200)].value` should return 
```
[ 250, 300, 350]
```
but current returning 
```
[[250], [300, 350]]
```
This result is happening because there is no flattening for `range` or `filter` operation

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: ci
+on: push
```
Comment (pradithya) [inline .github/workflows/test.yaml]: Should be on pull request as well

**Comment by pradithya** [comment]
Can you remove `parse_filter_v1`? It seems a dead code.

File: .github/workflows/test.yaml (hunk 1 above)
Comment (tiopramayudi) [reply .github/workflows/test.yaml]: updated

**Comment by tiopramayudi** [comment]
> Can you remove `parse_filter_v1`? It seems a dead code.

done

**Review by pradithya — APPROVED** [review]
LGTM

**Pull request merged**